src/components/cylc/WarningIcon.vue

```diff
@@ -0,0 +1,122 @@
+<!--
+    WarningIcon - A dismiss-able warning icon
+-->
+
+<template>
+  <span
+    class="c-warn"
+    style="display:inline-block; vertical-align:middle"
+  >
+    <v-tooltip
+      bottom
+      :disabled="!message"
+    >
+      <template
+        v-slot:activator="{ on }"
+      >
+        <svg
+          viewBox="0 0 100 100"
+          v-on="on"
+          @click="deactivate"
+        >
+          <path
+            :d="path()"
+            :stroke-width="strokeWidth()"
+          v-bind:class="{'active': active}"
+          />
+        </svg>
+      </template>
+      <span>{{ message }}</span>
+    </v-tooltip>
+  </span>
+</template>
+
+
+<script>
+/* stuff we want to do once, when the component is loaded, rather than
+ * every time it is used. */
+// stroke
+const sw = 10 // stroke width
+const hsw = sw / 2 // half stroke width
+
+// bounding box
+const x1 = 2
+const x2 = 98
+const y1 = 8
+const y2 = 92
+
+// path commands
+const PATH = pathJoin([
```

Comment:
SVG has no primitive for triangles, drawing them is surprisingly hard as the stroke exceeds the bounding box on the corners. I've done it like this to facilitate changing appearance e.g. stroke-width, rounded corners, etc.